rosmon_core/src/main.cpp

```diff
@@ -326,10 +323,33 @@ int main(int argc, char** argv)
 
 		// Disable direct logging to stdout
 		ros::console::backend::function_print = nullptr;
+                
+                std::string logFile; 
 
 		// Open logger
-		if(logFile.empty())
+		if(const char* logDir = std::getenv("ROSMON_LOG_PATH"))
 		{
+                        std::string dir(logDir);
+                        dir = dir + "/rosmon";
+                        if (chdir(dir.c_str()) == 0 || mkdir(dir.c_str(), 0777) == 0) 
+                        {
+                                dir = dir + "/roslogs";
+                                if (chdir(dir.c_str()) == 0 || mkdir(dir.c_str(), 0777) == 0) 
+                                {
+                                        logFile = dir + "/" + launchInfo.launch_group + "_" + launchInfo.launch_config + "_" + launchInfo.robot_name + ".log";
```

Comment:
@gauthamm I tried to use node name by using ros::this_node::getName() as well as config->rosmonNodeName() but the string they returned was 'empty'. Also when looking at the log files, it appears a log file is only generated for the node launched by mon launch, not all the nodes launched by that node.

Reply:
Also I tested launching two same nodes and the log files did not appear to be effected by it.

Reply:
https://github.com/rapyuta-robotics/rosmon/blob/rr-devel/rosmon_core/src/ros_interface.cpp#L49